Sources/WhisperKit/Core/WhisperKit.swift

```diff
@@ -8,15 +8,24 @@ import Foundation
 import Hub
 import TensorUtils
 import Tokenizers
+import OSLog
+
+private let logger = Logger(
+    subsystem: Constants.Logging.subsystem,
+    category: "WhisperKit"
+)
+private let signposter = OSSignposter(logger: logger)
 
 @available(macOS 13, iOS 16, watchOS 10, visionOS 1, *)
 public protocol Transcriber {
+    func transcribe(audioPaths: [String], decodeOptions: DecodingOptions?, callback: TranscriptionCallback) async -> [String: Result<TranscriptionResult, Swift.Error>]
+    func transcribe(audioArrays: [[Float]], decodeOptions: DecodingOptions?, callback: TranscriptionCallback) async -> [Result<TranscriptionResult, Swift.Error>]
     func transcribe(audioPath: String, decodeOptions: DecodingOptions?, callback: TranscriptionCallback) async throws -> TranscriptionResult?
     func transcribe(audioArray: [Float], decodeOptions: DecodingOptions?, callback: TranscriptionCallback) async throws -> TranscriptionResult?
 }
 
 @available(macOS 13, iOS 16, watchOS 10, visionOS 1, *)
-public class WhisperKit: Transcriber {
+public actor WhisperKit: Transcriber {
```

Comment:
Curious, why this change?

Reply:
`WhisperKit` contains shared mutable state (e.g. `currentTimings` property), since this PR introduces possibility to run inference on multiple files in parallel, mutating this state from different threads might lead to crashes. Wdyt?

Reply:
we could possibly make `TranscriptionTimings` an actor so it can be updated from the multiple threads and leave `WhisperKit` as a class

Reply:
solved by introducing `TranscribeTask` class which is responsible for transcribing single chunk of audio